operator/cilium_node.go

```diff
@@ -370,6 +370,21 @@ func runCNPNodeStatusGC(name string, clusterwide bool, nodeStore cache.Store) {
 					}
 
 					for _, cnp := range cnpItemsList {
+						// If CNP status updates have been disabled, we clean all the
+						// possible updates written when the option was still enabled.
+						// This is done to avoid the stale updates to accumulate and thus
+						// hinder scalability.
+						if option.Config.DisableCNPStatusUpdates {
```

Comment:
This entire function, the `runCNPNodeStatusGC`, will only be executed if `operatorOption.Config.CNPNodeStatusGCInterval != 0 `. We advise the users to set `operatorOption.Config.CNPNodeStatusGCInterval` to `0` if they are running `option.Config.DisableCNPStatusUpdates=true`. This effectively means that this code won't be executed and the node status won't be removed.

Reply:
> At least the `install/kubernetes: add patch to ciliumnetworkpolicies/status` should be opened in a separate PR so that we can backport that change into all branches.

Moved to https://github.com/cilium/cilium/pull/20373